src/Kiota.Builder/Writers/Php/CodeMethodWriter.cs

```diff
@@ -380,13 +380,14 @@ private void WriteRequestGeneratorBody(CodeMethod codeElement, RequestParams req
         private void WriteDeserializerBody(CodeClass parentClass, LanguageWriter writer, CodeMethod method) {
             var inherits = parentClass.StartBlock?.Inherits != null;
             var fieldToSerialize = parentClass.GetPropertiesOfKind(CodePropertyKind.Custom);
+            writer.WriteLine("$currentObject = $this;");
```

Comment:
the this capture in Java was required because in Java the object initializers capture the this. But I think it might be a behaviour really specific to Java itself, it might not be needed in PHP (like it's not needed in C#/TS/Go).

Reply:
Makes sense. Added that to be safe from undefined behavior. Will update the code later once I'm sure that there won't be any undefined behavior removing the varibale.